app/Http/Controllers/Group/NavigateStateController.php

```diff
@@ -1,11 +1,12 @@
 <?php
 
-namespace App\Http\Controllers;
+namespace App\Http\Controllers\Grouop;
 
+use App\Http\Controllers\Controller;
 use App\Models\Group;
 use Illuminate\Http\Request;
 
-class GroupNavigateStateController extends Controller
+class NavigateStateController extends Controller
```

Comment:
I'm confused, are you saying that I would simply use the session in the controller and in the view?

I thought it would be more fluent to have the controller use `$group->chooseYear($year)` and the view would use {{ $group->chosenYear() }}, instead of requiring session logic, which doesn't mean anything. Where do I put the function that creates the session key?

What am I missing here?

Reply:
I'm not saying you need to do this in this PR, but for the sake of all that is holy in the world, please plan to delete absolutely everything about what you're doing here and never speak of it again... 😢

Reply:
What this is doing is letting the user save the preferred year and state, for the lifetime of the session, for each group they are working in. 

The reason for the key is not a route. It's just a unique name for the key in the user's session.